std/container/array.d

```diff
@@ -1014,13 +1021,46 @@ $(D r)
 
 @system unittest
 {
-    Array!int a;
+    struct Dumb { int x = 5; }
```

Comment:
I wanted to test a sequence of operations checking various aspects of reallocation. Do you think it'll be better to split it up?